plugins/org.eclipse.fordiac.ide.deployment.opcua/src/org/eclipse/fordiac/ide/deployment/opcua/OPCUADeploymentExecutor.java

```diff
@@ -572,20 +574,83 @@ public List<org.eclipse.fordiac.ide.deployment.devResponse.Resource> queryResour
 
 	@Override
 	public Response queryFBType(final FBTypeEntry entry) throws DeploymentException {
-		// TODO implement when the API and infrastructure is fully tested
-		throw new DeploymentException("Query FB Type not yet supported!"); //$NON-NLS-1$
+		final CallMethodRequest request = new CallMethodRequest(Constants.MGMT_NODE, Constants.QUERY_FB_TYPE_NODE,
+				new Variant[] {});
+		try {
+			final String message = MessageFormat.format(Constants.QUERY_FB_TYPE, getTypeNameWithHash(entry));
+			final CallMethodResult result = sendREQ("", request, message).get(); //$NON-NLS-1$
+			final Response response = parseResponse(result, Constants.QUERY_RESPONSE);
```

Comment:
We also need some changes in the response format, as `QUERY_RESPONSE` is expected to be used with `queryResources`. I would suggest to change the name of this variable to `QUERY_RESOURCES_RESPONSE` and to introduce a new variable to use for the new query methods.

```java
public static final String QUERY_TYPE_RESPONSE = "<Response>\n  {0}\n</Response>"; //$NON-NLS-1$
```

```suggestion
			final Response response = parseResponse(result, Constants.QUERY_TYPE_RESPONSE);
```